lib/charms/mongodb/v1/mongodb_tls.py

```diff
@@ -302,21 +314,52 @@ def _on_certificate_expiring(self, event: CertificateExpiringEvent) -> None:
 
         self.set_tls_secret(internal, Config.TLS.SECRET_CSR_LABEL, new_csr.decode("utf-8"))
 
-    def _get_sans(self) -> List[str]:
+    def get_new_sans(self) -> Dict:
         """Create a list of DNS names for a MongoDB unit.
 
         Returns:
             A list representing the hostnames of the MongoDB unit.
         """
+        sans = {}
+
         unit_id = self.charm.unit.name.split("/")[1]
-        return [
+        sans[SANS_DNS_KEY] = [
```

Comment:
nit: you could build the dictionary directly with both `SANS_DNS_KEY` and `SANS_IPS_KEY` instead of diving them into 3 steps